### ISSUE TITLE
[QUESTION] Dynamically add/remove tracks

### ISSUE THREAD

**Issue by juskuc** [flutter-webrtc/flutter-webrtc#1473]
First of all - thanks for the work you've put in creating this library!

I'm working on a mobile application using flutter-webrtc and I've mostly gotten things to work how I want them to, except for one thing. Let's say I make an audio call (I get user media with only audio constraint set to true), but I want the users to have the ability to turn on their camera mid-call, essentially turning it into a audio+video call. I need this to happen instantly, on-demand and without the process of renegotiation. One of the solutions for this problem that I've found was to get user media constraints with both audio and video, but disabling/enabling the video track as needed. This does the job, however when getting the user media while in an audio call, the hardware's camera usage light comes up, which isn't ideal. I've been messing around with transceivers/senders, looking at examples, but haven't found any success. I saw some people using dummy tracks and then replacing them with actual ones in JS, but I can't seem to find a way to create a dummy video track in flutter.

TLDR; I want to only capture audio stream when making an audio call, but add the video track mid-call without the process of renegotiation. Is it possible and if yes, how would I go about achieving that? Thanks!

**Comment by ash-hashtag** [comment]
you just have to renegotiate, after adding or removing tracks, send offer and handle answer again. That's the only way as far as I know

**Comment by juskuc** [comment]
> or you can send both video and audio during first negotiation, and immediately disable video, without user noticing and enable it when needed, but the users can see the camera permission being active always tho, while not capturing too.

Yeah, that's kind of what I've gone with but as I mentioned, it's not ideal - why would the app need camera permission and why would it be accessed during a call that's audio-only initially (and might stay that for the rest of the call's duration). 

I was wondering if it's possible to only access camera on demand? Perhaps if there was an option to send a dummy video track and later, when the user clicks the video button mid-call, I could capture video stream and switch the dummy track with the user's actual video track, making it essentially a seamless transition and avoid the long-running process of renegotiation.

Here's a short example by Jan-Ivar of what I'm trying to achieve: https://blog.mozilla.org/webrtc/warm-up-with-replacetrack/, which tells me it is possible, however, I haven't yet found a way to create a dummy video track in Flutter 😕 So just wondering if it's a limitation of the `flutter-webrtc` library, or it indeed is possible and it's just my lack of knowledge.
